### PR TITLE
Update /maker/setup page for micro:bit launch

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR updates the 'maker/setup' page for the upcoming micro:bit launch scheduled for April 4, 2023 - [link to adhoc instance](https://adhoc-alice-update-maker-text-studio.cdn-code.org/maker/setup)

This is the [micro:bit staging <> deploy <> launch doc](https://docs.google.com/document/d/1wu_SadRXOFKWUgma2LWUVKOvxW0L8gcSUdsbPTAsTZY/edit#heading=h.r845ez41xs57). This PR is scheduled to merge to staging on Thursday, 3/30.

It does the following:
- remove the micro:bit experiment flag,
- updates the text so that both micro:bit and Circuit Playground (CP) are referred to,
- adds Amplitude analytics,
- refactors the `SetupChecklist.jsx`.

Currently, users can use the micro:bit on production with the experiment flag 'enableExperiments=microbit'.
However, even with the experiment on, the text on the page under the WebSerial section only refers to the CP. 

BEFORE:

Screen Shot 2023-03-23 at 3 44 14 PM

This PR removes the micro:bit experiment flag, and updates this section. In addition the page title and other sections of the page are refreshed including placing WebSerial above the Maker App section.

AFTER:
Screen Shot 2023-03-23 at 3 45 39 PM

Screen Shot 2023-03-23 at 3 46 24 PM

Product requested that we add Amplitude analytics on the maker setup page. The following events are being tracked:
- when a user clicks on the 'Connect to Board' button under WebSerial to go through the setup checker, which type of board are they using: micro:bit, CP Classic, or CP Express,
- when a user updates their micro:bit Firmata, which version micro:bit are they using (v1 or v2),
- when a user attempts to update their micro:bit Firmata and the Firmata update fails.

I added these events to Amplitude [here](https://data.amplitude.com/code-org/Code.org%20Tracking%20Plan/events/main/latest).

This PR also updates the `SetupChecklist.jsx` file so that sub-render functions are used to simplify the `render` function of the component.

## Links
- [jira - [Micro:bit] Update final text on the setup connection page](https://codedotorg.atlassian.net/browse/SL-36)
- [jira - [Micro:bit] Remove micro:bit experiment flag and refactor](https://codedotorg.atlassian.net/browse/SL-655)
- [jira - [Micro:bit] Add analytics to Maker page](https://codedotorg.atlassian.net/browse/SL-705)

## Testing story
I tested locally and also created an adhoc instance so others could check on different devices using different boards: [https://adhoc-alice-update-maker-text-studio.cdn-code.org/maker/setup](https://adhoc-alice-update-maker-text-studio.cdn-code.org/maker/setup).
I updated unit tests in `SetupGuideTest.js` since the micro:bit experiment flag was removed and added a unit test in `SetupChecklistTest.js` to check for analytics.
Locally, I also tested that events were being sent to Amplitude by changing `ALWAYS_SEND` to `true` and using the staging `cdo_amplitude_api_key` - [instructions](https://docs.google.com/document/d/1uuo-vJs41SvWRokVHGu8trZr-h6axq6BHqe6fbjlRbk/edit#).

## Deployment strategy
This is the [micro:bit staging <> deploy <> launch doc](https://docs.google.com/document/d/1wu_SadRXOFKWUgma2LWUVKOvxW0L8gcSUdsbPTAsTZY/edit#heading=h.r845ez41xs57). This PR is scheduled to merge to staging on Thursday, 3/30.
## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/kits/maker/ui/SetupGuide.jsx
Comment (sanchitmalhotra126) [inline apps/src/lib/kits/maker/ui/SetupGuide.jsx]: This may be out of scope - but it looks like the reason some of these props are being marked as not required is because we're trying to use the `DescriptionCard` component as a kind of header component, right? I wonder if it would be worth creating a separate header card component or something that only had a title and a description (no image) for clarity's sake

**Review by sanchitmalhotra126 — APPROVED** [review]
One suggestion, but otherwise LGTM! It also may be worth pulling in @code-dot-org/teacher-tools to take a look at the Analytics-related changes to make sure everything looks good there.

File: apps/src/lib/kits/maker/boards/microBit/MBFirmataUpdater.js
Comment (bethanyaconnor) [inline apps/src/lib/kits/maker/boards/microBit/MBFirmataUpdater.js]: I am very slightly concerned about a colon in a key. I think it'll be okay but this is the kind of thing that can lead to parsing errors. I'd recommend testing this PR on staging after its merged (staging actually logs to Amplitude) to double check there aren't any issues

**Review by bethanyaconnor — APPROVED** [review]
Approving to make it clear there's nothing blocking from me -- I didn't read through most of this PR though so I'll defer to Sanchit (or any other reviewers) for merge approval!

File: apps/src/lib/kits/maker/boards/microBit/MBFirmataUpdater.js
Comment (fisher-alice) [reply apps/src/lib/kits/maker/boards/microBit/MBFirmataUpdater.js]: It's definitely not a problem to remove the colon so will remove - thanks for the heads up on that. I also am about to post a question in Slack #teacher-tools about being able to test before merging to staging. Thank you!

**Review by sanchitmalhotra126 — APPROVED** [review]

File: apps/src/lib/kits/maker/ui/SetupGuide.jsx
Comment (fisher-alice) [reply apps/src/lib/kits/maker/ui/SetupGuide.jsx]: Sounds good! Added a `HeaderCard` to `SetupGuide.jsx` and re-instated `DescriptionCard` props requirements - thanks!

File: apps/src/lib/kits/maker/boards/microBit/MBFirmataUpdater.js
Comment (fisher-alice) [reply apps/src/lib/kits/maker/boards/microBit/MBFirmataUpdater.js]: And thanks for sharing this [Amplitude for Developers](https://docs.google.com/document/d/1uuo-vJs41SvWRokVHGu8trZr-h6axq6BHqe6fbjlRbk/edit#) via Slack. I added a unit test for analytics based on the doc suggestion.

**Comment by fisher-alice** [comment]
> One suggestion, but otherwise LGTM! It also may be worth pulling in @code-dot-org/teacher-tools to take a look at the Analytics-related changes to make sure everything looks good there.

Thanks for tagging teacher-tools!

**Pull request merged**
